sdk/servicebus/Azure.Messaging.ServiceBus/src/Client/ServiceBusClientOptions.cs

```diff
@@ -49,6 +51,23 @@ public ServiceBusRetryOptions RetryOptions
                 _retryOptions = value;
             }
         }
+
+        /// <summary>
+        /// The list of plugins for the client.
+        /// </summary>
+        internal List<ServiceBusPlugin> Plugins { get; set; } = new List<ServiceBusPlugin>();
+
+        /// <summary>
+        /// Register a plugin to be used to alter
+        /// incoming/outgoing messages.
+        /// </summary>
+        /// <param name="plugin">The plugin instance to register.</param>
+        public void AddPlugin(ServiceBusPlugin plugin)
```

Comment:
Should we consider a `RemovePlugin` method, just in case?  I don't see it being used extensively, but the only way to get one out in the current form would be to create a new set of options and start over.   If you decide to make the `Plugins` set public, its definitely not needed, but may be nice just for customer expectations of symmetry with the API.   I'm more likely to use a `RemovePlugin` method if I added something with an `AddPlugin` method than I am to think about manually adjusting the set through the property.

Reply:
I don't think it is needed as you can just create a new options instance. The property is not exposed so the only way to add plugins is using `AddPlugin`. This is similar to adding pipeline policies in `ClientOptions`.

Reply:
Plugins are not dynamic and are an opt-in model.